test/aqua/operators/test_op_construction.py

```diff
@@ -363,6 +364,29 @@ def test_circuit_compose_register_independent(self):
 
         self.assertEqual(composed.num_qubits, 2)
 
+    def test_list_op_to_circuit(self):
+        """
+        unitary composed/tensored/summed operator should transpile to circuit
+        non-unitary operator should raise Exception
+        """
+        i = MatrixOp([[1, 0], [0, 1]])
+
+        unitary = (i ^ X ^ i) @ (Z ^ i ^ Y)
+        unitary.to_circuit()
+
+        non_unitary = X + Y + Z
+        # pylint: disable=unnecessary-lambda
+        self.assertRaises(ExtensionError, lambda: non_unitary.to_circuit())
+
+        # more general test case
+        u2 = MatrixOp(primitive=unitary_group.rvs(2))
+        u4 = MatrixOp(primitive=unitary_group.rvs(4))
+        u8 = MatrixOp(primitive=unitary_group.rvs(8))
+        c2 = u2.to_circuit_op()
+
+        op = ((X ^ u4) @ (Z ^ c2 ^ Y) @ u8) ^ u2
+        op.to_circuit()
```

Comment:
There should be a test statement of some sort here, maybe you could check the unitary that's produces by the circuit (with `quantum_info.Operator`) or check some other properties of the resulting circuit

Reply:
I just pushed new test.
It asserts if the circuit is equivalent to unitary produced by numpy operations only.